src/fsharp/ast.fs

```diff
@@ -1365,18 +1366,18 @@ and SynModuleDecls = SynModuleDecl list
 and 
     [<NoEquality; NoComparison>]
     SynExceptionSig = 
-    | SynExceptionSig of SynExceptionDefnRepr * SynMemberSigs * range
+    | SynExceptionSig of SynExceptionDefnRepr * SynMemberSigs * range:range
 
 and
     [<NoEquality; NoComparison; RequireQualifiedAccess>]
     SynModuleSigDecl =
-    | ModuleAbbrev      of Ident * LongIdent * range
-    | NestedModule      of SynComponentInfo * isRec : bool * SynModuleSigDecls * range
-    | Val               of SynValSig * range
-    | Types             of SynTypeDefnSig list * range
-    | Exception         of SynExceptionSig * range
-    | Open              of LongIdent * range
-    | HashDirective     of ParsedHashDirective * range
+    | ModuleAbbrev      of ident:Ident * longId:LongIdent * range:range
+    | NestedModule      of SynComponentInfo * isRecursive:bool * SynModuleSigDecls * range:range
+    | Val               of SynValSig * range:range
```

Comment:
After looking at this I wonder if we can change the type name to Range with uppercase R